meta/client.go

```diff
@@ -0,0 +1,216 @@
+package meta
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/XiaoMi/pegasus-go-client/idl/base"
+	"github.com/sirupsen/logrus"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/XiaoMi/pegasus-go-client/session"
+	"github.com/bluele/gcache"
+	"github.com/go-zookeeper/zk"
+)
+
+// TODO(jiashuo) store config file
+var zkAddrs = []string{""}
+var zkTimeOut = 1000000000 // unit ns, equal 1s
+var zkRoot = "/pegasus-cluster"
+var zkWatcherCount = 1024
+
+var globalClusterManager *ClusterManager
+
+type ClusterManager struct {
+	Mut    sync.RWMutex
+	ZkConn *zk.Conn
+	// table->TableInfoWatcher
+	Tables gcache.Cache
+	// metaAddrs->metaManager
+	Metas map[string]*session.MetaManager
+}
+
+// zkContext cancels the goroutine that's watching the zkNode, when the watcher
+// is evicted by LRU.
+type zkContext struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+type TableInfoWatcher struct {
+	tableName   string
+	clusterName string
+	metaAddrs   string
+	event       <-chan zk.Event
+	ctx         zkContext
+}
+
+// TODO(jishuo1) change log module
+func initClusterManager() {
+	zkConn, _, err := zk.Connect(zkAddrs, time.Duration(zkTimeOut))
+	if err != nil {
+		logrus.Panicf("failed to connect to zookeeper \"%s\": %s", zkAddrs, err)
+	}
+
+	tables := gcache.New(zkWatcherCount).LRU().EvictedFunc(func(key interface{}, value interface{}) {
+		value.(*TableInfoWatcher).ctx.cancel()
+		logrus.Warnf("table [%s] evicted from table cache (capacity: %d)", key.(string), zkWatcherCount)
+	}).Build() // TODO(jiashuo1) consider set expire time
+	globalClusterManager = &ClusterManager{
+		ZkConn: zkConn,
+		Tables: tables,
+		Metas:  make(map[string]*session.MetaManager),
+	}
+}
+
+func (m *ClusterManager) getMeta(table string) (*session.MetaManager, error) {
+	var meta *session.MetaManager
+
+	tableInfo, err := globalClusterManager.Tables.Get(table)
+	if err == nil {
+		meta = globalClusterManager.Metas[tableInfo.(*TableInfoWatcher).metaAddrs]
+		if meta != nil {
+			return meta, nil
+		}
+	}
+
+	logrus.Printf("[%s] can't get cluster info from local cache, try fetch from zk.", table)
+	globalClusterManager.Mut.Lock()
```

Comment:
```suggestion
	m.Mut.Lock()
```

replace `m` with `globalClusterManager`